src/js/time-manager.js

```diff
@@ -59,36 +68,54 @@ class TimeManager extends LitElement {
   }
 
   async setWorkedHours() {
+    // set up
     const storedTimes = await getTimePairs('time-pairs-only');
     const nowDecimal = human2decimalTime();
-    let newWorkedHours = 0
+    let newWorkedHours = 0;
+    let lastStopTime = 0;
+    let pauses = 0;
+    // for all time-interval pairs:
+    storedTimes.forEach(([startTime, stopTime], i) => {
+      // valid start time?
+      if (isNaN(startTime) || startTime > nowDecimal) {
+        // no, so skip this time interval
+        return;
+      }
+      // for first time interval only, ...
+      if (i === 0) {
+        // ... remember it as start-of-day
+        this.dayStart = startTime;
+        // ... and initialize pseudo last-stop-time
+        lastStopTime = startTime;
+      }
 
-    storedTimes.forEach(([startTime, stopTime]) => {
-      if(startTime) {
-        const stopTimeIsNaN = isNaN(stopTime) || stopTime === undefined;
+      // ensure valid stop time
+      stopTime = isNaN(stopTime)
+        ? /* unfinished time interval */ nowDecimal
+        : stopTime;
 
-        if(stopTimeIsNaN) {
-          //simulate endTime = now
-          stopTime = nowDecimal;
-        }
+      // add time interval length to worked hours sofar
+      newWorkedHours += stopTime - startTime;
 
-        // do not count if given startTime is before now
-        if(startTime <= nowDecimal) {
-          newWorkedHours += stopTime - startTime;
-        }
-      }
+      // calculate pause increment
+      pauses += startTime - lastStopTime;
+      // lag stop time
+      lastStopTime = stopTime;
     });
-
+    // transfer to instance variables
     this.workedHours = newWorkedHours;
+    this.pauses = pauses;
   }
 
   setEndtime() {
-    dailyHours().then((hours) =>{
-      const hoursLeftDecimal = hours - this.workedHours;
-      const nowDecimal = human2decimalTime();
-
-      this.endTime = nowDecimal + hoursLeftDecimal;
-    } )
+    // get prescribed daily hours (from settings)
+    dailyHours().then(hours => {
+      // calculate end-of-work time
+      // N.B. Avoiding reference to current time ('now') in the calculation
+      // prevents wrong results in case of on-device clock skew
```

Comment:
Ist das ein reales problem? Kann ich mir nicht vorstellen

Reply:
Yes, it is. It was prompted by my dev laptop being on wrong time due to NNTP server inaccessibility or something, which happens with some regularity.
A time tracker must be incredibly robust to be really trusted by a wider audience... this is an important contribution to this goal.